src/main/java/com/getindata/connectors/http/internal/sink/httpclient/status/ComposeHttpStatusCodeChecker.java

```diff
@@ -0,0 +1,128 @@
+     * @return true if status code is considered as error or false if not.
+     */
+    public boolean isErrorCode(int statusCode) {
+
+        Preconditions.checkArgument(
+            statusCode >= MIN_HTTP_ERROR_CODE,
+            String.format(
+                "Provided invalid Http status code %s,"
+                    + " status code should be equal or bigger than 100.",
+                statusCode)
+        );
+
+        boolean isWhiteListed = excludedCodes.stream()
+            .anyMatch(check -> check.isWhiteListed(statusCode));
+
+        if (isWhiteListed) {
+            return false;
+        } else {
+            return errorCodes.stream()
+                .anyMatch(httpStatusCodeChecker -> httpStatusCodeChecker.isErrorCode(statusCode));
+        }
+    }
+
+    private void prepareErrorCodes(Properties properties) {
+        String sErrorCodes =
+            properties.getProperty(HttpConnectorConfigConstants.HTTP_ERROR_CODES_LIST, "");
+
+        if (StringUtils.isNullOrWhitespaceOnly(sErrorCodes)) {
+            errorCodes.addAll(DEFAULT_ERROR_CODES);
+        } else {
+            String[] splitCodes = sErrorCodes.split(HttpConnectorConfigConstants.ERROR_CODE_DELIM);
+
+            if (splitCodes.length == 0) {
+                errorCodes.addAll(DEFAULT_ERROR_CODES);
+            } else {
+                prepareErrorCodes(splitCodes);
+            }
+        }
+    }
+
+    /**
+     * Process given array of status codes and assign them to
+     * {@link SingleValueHttpStatusCodeChecker} for full codes such as 100, 404 etc. or to
+     * {@link TypeStatusCodeChecker} for codes that were constructed with "XX" mask
+     */
+    private void prepareErrorCodes(String[] statusCodes) {
+        for (String sCode : statusCodes) {
+            if (!StringUtils.isNullOrWhitespaceOnly(sCode)) {
+                String trimCode = sCode.toUpperCase().trim();
+                boolean isTypeCode = isTypeCode(trimCode);
+                if (isTypeCode) {
+                    int code = Integer.parseInt(trimCode.replace("X", ""));
+                    errorCodes.add(new TypeStatusCodeChecker(HttpType.getByCode(code)));
+                } else {
+                    errorCodes.add(
+                        new SingleValueHttpStatusCodeChecker(Integer.parseInt(trimCode))
+                    );
+                }
+            }
+        }
+    }
+
+    private Set<WhiteListHttpStatusCodeChecker> prepareWhiteList(Properties properties) {
+        return Arrays.stream(
+                properties.getProperty(HttpConnectorConfigConstants.HTTP_ERROR_CODE_WHITE_LIST, "")
+                    .split(HttpConnectorConfigConstants.ERROR_CODE_DELIM))
+            .filter(sCode -> !StringUtils.isNullOrWhitespaceOnly(sCode))
+            .map(String::trim)
+            .mapToInt(Integer::parseInt)
+            .mapToObj(WhiteListHttpStatusCodeChecker::new)
+            .collect(Collectors.toSet());
+    }
+
+    boolean isTypeCode(final String code) {
```

Comment:
you are right.
and actually this method can be simplified if we assume that code argument will be upper case, trim and 3 characters long which will be true because there is validation for that earlier.

I've pushed changes to this method + some tests.

Reply:
Ok, LGTM. However, I am thinking out loud, can we have an assertion about length of the `code` in this method too? Although I know it is checked before `prepareErrorCodes` callsite, one might call this function from the outside (especially considering it is an implementation of `HttpStatusCodeChecker` interface) and would yield wrong results because of it.

Reply:
I would really like to avoid adding this size check into this method.
Nice validation with validation messages is a big rabbit hole that we can explore in the future I think.

Let's do this:
I made isTypeCode private, I added couple test cases to shouldThrowOnInvalidCodeRange + I'm hoping that people will read Java doc for this method which says:
"This method expects that provided string will be 3 elements long, trim and upper case."